utils/directories.cc

```diff
@@ -77,9 +77,11 @@ void directories::set::add_sharded(sstring p) {
     }
 }
 
-directories::directories(bool developer_mode)
-        : _developer_mode(developer_mode)
-{ }
+directories::directories(const ::db::config& cfg)
+        : _developer_mode(cfg.developer_mode())
+{
+    setup_paths(cfg);
```

Comment:
I don't think this call is needed. The setup_paths(cfg) does a bunch of `set_foo_dir(cfg.foo_directory())` calls and each `set_foo_dir()`, in turn, just initializes the corresponding `_foo_dir` member. It's better to initialize `_foo_dir`-s in constructor list-initializer directly

Reply:
Moved to the initializer list and removed setters entirely.